lib/gitWorkflow.js

```diff
@@ -186,24 +186,14 @@ class GitWorkflow {
        */
       if (!shouldBackup) return
 
-      // Get a list of unstaged deleted files, because certain bugs might cause them to reappear:
-      // - in git versions =< 2.13.0 the `--keep-index` flag resurrects deleted files
-      // - git stash can't infer RD or MD states correctly, and will lose the deletion
-      this.deletedFiles = await this.getDeletedFiles()
-
-      // the `git stash` clears metadata about a possible git merge
-      // Manually check and backup if necessary
+      // When backup is enabled, the revert will clear ongoing merge status.
       await this.backupMergeStatus()
 
-      // Save stash of original state
-      await this.execGit(['stash', 'save', STASH])
-      await this.execGit(['stash', 'apply', '--quiet', '--index', await this.getBackupStash()])
-
-      // Restore meta information about ongoing git merge, cleared by `git stash`
-      await this.restoreMergeStatus()
```

Comment:
No need to restore merge status here, since `git stash create` didn't remove it. The backup on line 196 is still required for the revert when tasks fail (because of `git reset`).